.github/workflows/reusable-windows-ci.yml

```diff
@@ -0,0 +1,769 @@
+          Import-Module $pwshmodule
+          Enter-VsDevShell -SetDefaultWindowTitle -VsInstallPath $pwshmodule.Directory.Parent.Parent `
+                            -StartInPath (pwd) -DevCmdArguments '/arch=x64 /host_arch=x64'
+
+          cd $deploy_dir
+
+          nmake .\Makefile.mak OPENSSL_DIR=$Env:OPENSSL64_ROOT BUILD_FOR=WIN64
+          $config = ls -Path $Env:OPENSSL64_ROOT -Recurse -Include openssl.cnf;
+          $libp11_path = Join-Path $deploy_dir src
+          'LibP11_ROOT_64=' + $libp11_path | Out-File $Env:GITHUB_ENV -Append -Encoding OEM
+
+          # Wait until installation is completed
+          Wait-Process -Name msiexec -ErrorAction SilentlyContinue
+          del $msiHSM
+
+          # Set up OpenSSL
+          $module_path = gci -Path $Env:SOFTHSM2_ROOT -Recurse -Include '*.dll' | ? FullName -match 64
+          $contents = gc $config;
+
+          $header = "# HSM test ancillary configuration",
+                    "openssl_conf = openssl_init"
+
+          $footer = "[openssl_init]",
+                    "engines = engine_section",
+                    "",
+                    "[engine_section]",
+                    "pkcs11 = pkcs11_section",
+                    "",
+                    "[pkcs11_section]",
+                    "engine_id = pkcs11",
+                    ("dynamic_path =" + ("$libp11_path/pkcs11.dll" -replace "\\","/")),
+                    ("MODULE_PATH =" + ($module_path.FullName -replace "\\","/")),
+                    "init = 0"
+
+          $header_line = ($contents | sls '^HOME')[0].LineNumber;
+          ($contents[0..$header_line] + $header +  $contents[$header_line..$contents.count] + $footer) |
+          % { $_.TrimStart() } | Out-File $config -Encoding OEM;
+
+          # lead openssl to the right config file
+          $Env:OPENSSL_CONF=$config
+
+          # introduce openssl binary in the path for the lower check
+          $Env:Path += ";$Env:OPENSSL64_ROOT\bin"
+
+          # check if is working
+          openssl engine pkcs11 -t
+
+      - name: Prepare colcon workspace
+        run: |
+          # Get some convenient tools
+          Install-Module -Name ConvertBase64Strings -Force -AllowClobber
+          Import-Module -name ConvertBase64Strings -Prefix CI
+          Install-Module powershell-yaml -Force
+
+          # Check if git is behaving
+          $checker = @{
+            Begin = {$errores = 0}
+            Process = {if($_ -is [Management.Automation.ErrorRecord] -and $_.Exception.Message -match "^(error|fatal):") {++$errores} }
+            End = { $errores -eq 0 }
+          }
+          $assert = {if($_ -is [Management.Automation.ErrorRecord] -and $_.Exception.Message -match "^(error|fatal):")
+                        { Write-Error $_ } else { Write-Output $_ } }
+
+          # Download Fast-DDS-CI repo and checkout current branch
+          git clone https://github.com/eProsima/Fast-DDS-CI.git Fast-DDS-CI 2>&1 | % $assert
```

Comment:
We have an alias in place and this works, but I think it would be better if we changed it to eprosima-ci